### PR TITLE
fix: balance missmatch

### PR DESCRIPTION
**Pull request by pandadefi**
## Description

- Fix end of slope not set _chainsecurity report_
- Fix total supply and user balance missmatch
- statemind: veYFI balance might not decrease
- chainsecurity: 5.3 Kink Timestamp Can Be Set to a Past Timestamp

There is still a little mismatch between the user balance and the total supply.

## Checklist

- [x] I have run vyper and solidity linting
- [x] I have run the tests on my machine
- [x] I have followed commitlint guidelines
- [x] I have rebased my changes to the latest version of the main branch

### CODE REVIEW

File: tests/functional/test_voting_escrow.py
Comment (storming0x) [reply tests/functional/test_voting_escrow.py:45]: I think adding an additional assert here for 

`assert ve_yfi.totalSupply() >= ve_yfi.balanceOf(alice)`

Should be also a good invariant to test, we never want to owe users balance more than our total supply

File: tests/functional/test_voting_escrow.py
Comment (storming0x) [inline tests/functional/test_voting_escrow.py:12]: can we also add a test for the minimum lock time to check balances and assumptions are correct for a user based on these latest changes?

**Review by storming0x — COMMENTED** [review]
Did an overall review, i think changes look good, ran the test in master to ensure the test failed without fixes and it does.

Small suggestions mainly, i think adding also a minimum lock test would cover both ranges of locking we want to make sure to validate.

File: tests/functional/test_voting_escrow.py
Comment (pandadefi) [reply tests/functional/test_voting_escrow.py:45]: That's only valid when this is also merged
https://github.com/yearn/veYFI/pull/181

File: tests/functional/test_voting_escrow.py
Comment (pandadefi) [reply tests/functional/test_voting_escrow.py:12]: Can you add it? I am trying to get more people involved in coding this here, this is a simple test, please do it.

Comment (pandadefi) [reply tests/functional/test_voting_escrow.py:12]: The test exists already check line 155 bob creates a lock for only a week

File: tests/functional/test_voting_escrow.py
Comment (pandadefi) [reply tests/functional/test_voting_escrow.py:45]: https://github.com/yearn/veYFI/pull/179/commits/ffc0d3813412b04fbc194612b54e8891308b3f97

File: tests/functional/test_voting_escrow.py
Comment (storming0x) [reply tests/functional/test_voting_escrow.py]: > Can you add it? I am trying to get more people involved in coding this here, this is a simple test, please do it.

I will finish helping with reviews first to prioritize getting context on code and audit changes first before i dive into adding tests, i will start helping on testing side to get more coverage

**Review by storming0x — APPROVED** [review]
Changes LGTM!, CI tests should be fixed once other PRs are merged together

**Pull request merged**
